tests/test_models.py

```diff
@@ -73,7 +73,7 @@ def my_own_save_function(filepath):
     clear_save_dir()
 
     # -----------------
-    # CASE K=1 (2.5, epoch 4)
+    # CASE K=1 (2.5, epoch 5)
```

Comment:
but it's defined as:  
```python
losses = [10, 9, 2.8, 5, 2.5]
```   
2.5 is the 4th epoch... so your tests are off by 1. this should be:  

```python
CASE K=1 (2.5, epoch 5)
```

This means your impelementation is off by 1